onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu

```diff
@@ -1,241 +0,0 @@
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-// Modifications: Add SkipLayerNormKernelVec to
-//                leverage vectorized load/write.
-//                and templatize ComputeSkipLayerNorm for different
-//                data types.
-// Copyright (c) Advanced Micro Devices, Inc. All rights reserved.
-// Licensed under the MIT License.
-
-#include "contrib_ops/cuda/bert/layer_norm.cuh"
-#include "contrib_ops/cuda/bert/skip_layer_norm_impl.h"
-#include <cuda_fp16.h>
-
-namespace onnxruntime {
-namespace contrib {
-namespace cuda {
-
-namespace {
-template <typename T>
-T maybe2half(float x);
-
-template <>
-float maybe2half(float x) {
-  return x;
-}
-
-template <>
-half maybe2half(float x) {
-  return __float2half_rn(x);
-}
-
-// Using only power of 2 numbers will lead to waste of compute for same size such as 768, which is a very common case
-// in BERT. Ideally we can step by wrap_size * num_unroll, but listing too many steps will cause long compile time.
-constexpr int kSizes[] = {32, 64, 128, 384, 768, 1024, 2048};
-constexpr int kMinBlockSize = 32;
-constexpr int kMaxBlockSize = 256;
-
-int NextSize(int x) {
-  size_t len = sizeof(kSizes) / sizeof(kSizes[0]);
-  for (size_t i = 0; i < len; ++i) {
-    if (x <= kSizes[i]) {
-      return kSizes[i];
-    }
-  }
-  return kSizes[len - 1];
-}
-
-template <typename T, int NumUnroll>
-bool CanVectorized(T* output, T* skip_input_bias_add_output, const T* input, const T* skip, const T* gamma,
-                   const T* beta, const T* bias, const int ld, const int next_size) {
-  constexpr int alignment = std::alignment_of<aligned_vector<T, NumUnroll>>::value;
-  return ld % NumUnroll == 0 && reinterpret_cast<uint64_t>(output) % alignment == 0 &&
-         reinterpret_cast<uint64_t>(skip_input_bias_add_output) % alignment == 0 &&
-         reinterpret_cast<uint64_t>(input) % alignment == 0 && reinterpret_cast<uint64_t>(skip) % alignment == 0 &&
-         reinterpret_cast<uint64_t>(gamma) % alignment == 0 && reinterpret_cast<uint64_t>(beta) % alignment == 0 &&
-         reinterpret_cast<uint64_t>(bias) % alignment == 0 && next_size / NumUnroll >= kMinBlockSize &&
-         next_size / NumUnroll <= kMaxBlockSize;
-}
-}  // namespace
-
-template <typename T, unsigned TPB, bool Simplified>
-__global__ void SkipLayerNormKernel(
-    const int ld, const T* input, const T* skip,
-    const T* beta, const T* gamma, const T* bias,
-    const T epsilon, T* output, T* skip_input_bias_add_output) {
-  const T reverse_ld = T(1.f / ld);
-  const int offset = blockIdx.x * ld;
-
-  KeyValuePairSum pair_sum;
-  // reduce x and x^2
-  cub::KeyValuePair<T, T> thread_data(0, 0);
```

Comment:
Should we check with the training team on whether it is okay to switch over completely from the existing implementation ? I think the training team introduced some optimizations in this file recently and they might still want to use this code path

Reply:
Add them to PR